pkg/controllers/federate/controller.go

```diff
@@ -86,6 +86,10 @@ type FederateController struct {
 	logger  klog.Logger
 }
 
+func (c *FederateController) IsControllerReady() bool {
```

Comment:
Let's create a `HasSynced` method and call it in both `IsControllerReady` and `Run` so the two are always in sync.  This should also apply to all the other controllers.